python/pyspark/pandas/frame.py

```diff
@@ -12212,7 +12237,8 @@ def explode(self, column: Name) -> "DataFrame":
         data_fields[idx] = field.copy(dtype=dtype, spark_type=spark_type, nullable=True)
 
         internal = psdf._internal.with_new_sdf(sdf, data_fields=data_fields)
-        return DataFrame(internal)
+        result_df: DataFrame = DataFrame(internal)
```

Comment:
Just curious: any helps of this annotaion? Because I saw many inconsist in current PS codebase.

```python
pdf = DataFrame(internal)
pdf: DataFrame = DataFrame(internal)
```

I thought annotation only helps in public interface/parameter or schema infer, so any other one should be cleanup.

cc @zero323